client/utils/guidance.py

```diff
@@ -0,0 +1,210 @@
+from client.protocols.common import models
+from client.utils import auth
+from urllib.request import urlopen
+from client.utils import assess_id_util
+import json
+import os
+import sys
+
+"""
+This utility is called by unlock.py. This guidance utility changes the message that students see
+after they input in wrong answers. If a student guesses a certain amount of wrong answers
+that all show a certain type of confusion about the problem, this utility will instead of showing
+the default "Not Quite Try Again" message will show some kind of message that will target 
+that type of misunderstanding. 
+
+This utility object requires internet access to determine what treatmeng group they are assigned
+to. The different treatment groups will have varying threshold level of answers as well as different
+messages and other differences. It will contact the server defined below in the variable TGSERVER with
+the user's email and lab assignment to get the treatment group number. 
+
+Commonly used acronyms: 
+TG = treatment group number
+KI = Type of targeted understanding
+misU = Type of misunderstanding the student is showing
+"""
+
+TGSERVER = "http://tg-server.app.cs61a.org/"
+LOCAL_TG = "tests/tg.ok_tg"
+OK_GUIDANCE = "tests/.ok_guidance"
+DEFAULT_MSG = "-- Not quite. Try again! --"
+DEFAULT_RETURN_VALUE = ({}, -1, "")
+COUNT_FILE_PATH = "tests/misUcount.json"
+TG_SERVER_ENDING = "/unlock_tg"
+TG_CONTROL = 0
+# If student forces guidance messages to show, we will assign treatment group number below
+GUIDANCE_FLAG_TG_NUMBER = 1
+TG_ERROR_VALUE = -1
+
+lambda_string_key_to_func = {
```

Comment:
Please add a comment describing what these lambdas do, to help developers who aren't familiar with this guidance project.